lib/embulk/guess/schema_guess.rb

```diff
@@ -62,11 +62,8 @@ def guess_type(str)
           return "long"
         end
 
-        if str.include?('.')
-          a, b = str.split(".", 2)
-          if a.to_i.to_s == a && b.to_i.to_s == b
-            return "double"
-          end
+        if str =~ /^[-+]?([1-9]\d*|0)(\.\d+)([eE][-+]?[0-9]+)?$/
```

Comment:
1. This regex is to make better suggestion for double type instead of current one. The current behavior has a problem, that it could not suggest float values includes 0 after decimal point as double type like 1.01, 5.001. It's ok because this is just suggestion. But, if we could improve it, it would be better. I could use a strict spec for floating point like IEEE 754 for my regex but, since it might take a time, I skipped the implementation for now. Actually finite numbers, infinities, NaNs defined in IEEE 754 are not supported in my regex but, should be supported in the future. I will put this comments on the source code later.

2. Thank you. I will fix bad codes later.

3. I could not find better libraries as the result of my survey. Float in Ruby and parseDouble in Java were candidates but, they can parse/convert "001", "002" like string as float/double types. Since I think that it's better to suggest "001" and "002" as string type in Embulk, I don't use them and use the regex. 

4. Yea, actually we don't have spec but, IEEE 754 might be one candidate.